### ISSUE TITLE
Export Results exporting special characters as Unicode Hex Character

### ISSUE THREAD

**Issue by vogtjob** [NUWCDIVNPT/stig-manager#1058]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Are you using the latest version of STIG Manager?

- [X] I am using the latest [Release](https://github.com/NUWCDIVNPT/stig-manager/releases).
- [ ] I am NOT using the latest [Release](https://github.com/NUWCDIVNPT/stig-manager/releases). I am aware that the first thing I will be asked to do is update the application so that I have the latest bugfixes.

### Where are you experiencing the issue?

- [X] API
- [ ] UI or other client
- [ ] Deployment
- [ ] Elsewhere

### Current Behavior

Using "Export Results" in the API and choosing CKL formats, exports data with unicode hex characters.  Below are a couple of examples.    Checking my configurations with past versions, the issue started occurring with the 1.3.7 release.

EX:
<STIG_DATA>
          <VULN_ATTRIBUTE>Fix_Text</VULN_ATTRIBUTE>
          <ATTRIBUTE_DATA>Trust must be established when utilizing Publishers Membership Condition.  All publishers&amp;#x27; certificates must have documented approvals from the IAO.</ATTRIBUTE_DATA>
        </STIG_DATA>

EX2: 
&amp;#x3C;application name=&amp;#x201C;remoteserver&amp;#x201D;&amp;#x3E; 
  &amp;#x3C;service&amp;#x3E; 
    &amp;#x3C;activated type=&amp;#x201C;sample.my.object, myobjects&amp;#x201D;/&amp;#x3E; 
  &amp;#x3C;/service&amp;#x3E; 
  &amp;#x3C;channels&amp;#x3E; 
    &amp;#x3C;channel ref=&amp;#x201C;http server&amp;#x201D; port=&amp;#x201C;80&amp;#x201D;/&amp;#x3E; 
  &amp;#x3C;/channels&amp;#x3E; 
&amp;#x3C;/application&amp;#x3E;

### Expected Behavior

Expect "export results" to export as human readable. 

### Describe your Environment

```markdown
- Hosting: Hosting
- Browser: Edge, Chrome
- OS: RHEL 8 using podman containers of stig-manager 1.3.8, mysql 8.0.33, nginx 1.25
```

**Comment by cd-rite** [comment]
HI @vogtjob 
I can confirm this behavior, however, I don't see it starting with 1.3.7. Did you migrate from 1.3.6 or were you using an earlier release before that?

Unfortunately, there are certain characters that must be escaped to create a valid XML document, and those are the ones I think you are seeing in our .ckl files.  Some of those characters have exceptions and can be left unescaped in certain contexts like element values, but there are sub-exceptions to that behavior as well! We will determine an LOE to see if it's possible to be a little more surgical when encoding those characters.

I think any xml viewer (for example: https://codebeautify.org/xmlviewer) should be resolving those characters into their human-readable form, it would only be when looking at the raw xml in a text editor that it would show the escaped characters. 

I can confirm this behavior, however, I don't see it starting with 1.3.7. Did you migrate from 1.3.6 or were you using an earlier release before that?

Thanks!

**Comment by vogtjob** [comment]
The issue was brought to me by one of our users who noticed the issue opening the exported .ckl's into STIG-Viewer (version didn't matter).  At the time he was exporting from STIG-Manager 1.3.8.   When opening the .ckl's the special characters appeared throughout. 

Your response made me look a little closer and, sure enough, characters are in 1.3.6 and previous versions .ckls, but they are in a different format and opened in a way that STIG-Viewer "liked".  For comparison I also opened up in Firefox and Edge and saw the same discrepancies with the characters (1.3.6 opened fine, newer versions showed the characters).  For the record codebeautify liked both formats. 

I've attached two dummy checklist's below.  One exported from 1.3.6, the other from 1.3.9.  They are from identical configurations other than the STIG-Manager version.  Just change the extension to .ckl.

An easy example to see is V-230234: The examples below show how it is rendered in STIG Viewer (The same as on GITHUB).

 **1.3.9:** 

<ATTRIBUTE_DATA>For systems that use BIOS, this is Not Applicable.

Check to see if an encrypted grub superusers password is set. On systems that use UEFI, use the following command:

$ sudo grep -iw grub2_password /boot/efi/EFI/redhat/user.cfg

GRUB2_PASSWORD=grub.pbkdf2.sha512.[password_hash]

If the grub superusers password does not begin with &amp;#x22;grub.pbkdf2.sha512&amp;#x22;, this is a finding.</ATTRIBUTE_DATA>

**1.3.6**

<ATTRIBUTE_DATA>For systems that use BIOS, this is Not Applicable.

Check to see if an encrypted grub superusers password is set. On systems that use UEFI, use the following command:

$ sudo grep -iw grub2_password /boot/efi/EFI/redhat/user.cfg

GRUB2_PASSWORD=grub.pbkdf2.sha512.[password_hash]

If the grub superusers password does not begin with &#x22;grub.pbkdf2.sha512&#x22;, this is a finding.</ATTRIBUTE_DATA>

Thank you for your time.


[test-RHEL_8_STIG-V1R7_1.3.9.txt](https://github.com/NUWCDIVNPT/stig-manager/files/12372114/test-RHEL_8_STIG-V1R7_1.3.9.txt)
[test-RHEL_8_STIG-V1R7_1.3.6.txt](https://github.com/NUWCDIVNPT/stig-manager/files/12372136/test-RHEL_8_STIG-V1R7_1.3.6.txt)

**Comment by cd-rite** [comment]
Hi @vogtjob Thanks for the additional info! It really helped clarify the issue. 
When we updated our xml library in 1.3.7, there was new behavior that was causing some characters to be double-escaped when configured the same way as the older version.

We have a PR with a fix: #1066 
Numbered release will likely happen next week, or you can grab our "latest" container after merge.
Thanks for the contribution!
